src/Compiler/CMakeLists.txt

```diff
@@ -66,9 +66,21 @@ add_onnx_mlir_library(CompilerPasses
   )
 
 if (ONNX_MLIR_VENDOR)
-  set_source_files_properties(CompilerUtils.cpp
-    PROPERTIES COMPILE_DEFINITIONS "ONNX_MLIR_VENDOR=\"${ONNX_MLIR_VENDOR} \"")
+  list(APPEND DEFINITIONS "ONNX_MLIR_VENDOR=\"${ONNX_MLIR_VENDOR}\"")
 endif()
+if (PRODUCT_VERSION)
```

Comment:
I would recommend keeping the naming convention and version specification consistent with LLVM e.g.:

```
if(NOT DEFINED LLVM_VERSION_MAJOR)
  set(LLVM_VERSION_MAJOR 15)
endif()
if(NOT DEFINED LLVM_VERSION_MINOR)
  set(LLVM_VERSION_MINOR 0)
endif()
if(NOT DEFINED LLVM_VERSION_PATCH)
  set(LLVM_VERSION_PATCH 0)
endif()
if(NOT DEFINED LLVM_VERSION_SUFFIX)
  set(LLVM_VERSION_SUFFIX git)
endif()
```

This sets a default version if none is specified, and also uses the major/minor/patch naming which is fairly standard.

Reply:
+1  for try to be consistent with LLBM naming conventions.

Assuming that the names are `PRODUCT_VERSION_MAJOR`, `PRODUCT_VERSION_MINOR`, `PRODUCT_VERSION_PATCH` and `PRODUCT_ID` I think that the defaults should be:
   - PRODUCT_VERSION_MAJOR = 0
   - PRODUCT_VERSION_MINOR = 0
   - PRODUCT_VERSION_PATCH = 0
   - PRODUCT_ID = "NOT_SPECIFIED"
  
Also, there should be some constraints on usage in the `cmake` files. I am thinking about these rules:
  - if  `PRODUCT_ID` is not set via cmake then overwriting `PRODUCT_VERSION_MAJOR` or `PRODUCT_VERSION_MINOR` or `PRODUCT_VERSION_PATCH` is an configuration error to be diagnosed by cmake
  - if `PRODUCT_ID` is set at least ``PRODUCT_VERSION_MAJOR` must be set to a positive integer value different that 0 
  - if `PRODUCT_VERSION_MINOR` is set then `PRODUCT_VERSION_MAJOR` must be set as well 
  - if `PRODUCT_VERSION_PATCH` is set then  `PRODUCT_VERSION_MINOR` must be set as well

Reply:
@sstamenova Thanks for the comments. I agree with the standard naming.
What's the benefit of having a default version if none is specified? The only one I can think of is so that we can create a test case. The disadvantage is we increase the LLVM file size with not useful metadata.

Reply:
Added cmake variables constraints as suggested by @etiotto .